### DISCUSSION TITLE
Passing the bootstrapper window handle to subsequent processes in Wix5

### DISCUSSION THREAD

**Discussion by tom-englert** [wixtoolset/issues#8171] [Questions]
I'm actually migrating my Wix BA to Wix5, and got stuck on this problem:

When upgrading an existing application, the bootstrapper process is started twice, and in the second process only `OnPlanComplete` is called.
Now from `OnPlanComplete` I call `engine.Apply(_windowHandle)`, however since this is a new process, and `Run` has never been called in this process, the window handle is zero, and `engine.Apply` fails with an argument exception.

I can pass the handle to the second process using e.g. shared memory - then repair and upgrade work.

However I wonder if checking the window handle in `engine.Apply` is really correct, e.g. during a silent install there won't be a window and thus no window handle.

**Comment by petm2022** [comment]
I had the same issue when upgrading to Wix4. My solution was to use EnsureHandel().

In my Model class I added a property:
`public IntPtr ViewWindowHandle { get; set; }
`

Then using this property in OnPlanComplete:
`Engine.Apply(this.ViewWindowHandle);`

In my Run function I assigned the propery:
`model.ViewWindowHandle = new WindowInteropHelper(view).EnsureHandle();
`

Hope this will help you.
